### PR TITLE
Backend integration release

### PR DESCRIPTION
**Pull request by abeiler7**
** This is the PR for the Release branch **

To start, this commit touches a lot of the files in the project. Many files just contain
       little changes to fix or cleanup issues that come up during integration.
A new function, logicCntrl(), was added to spell_check.py. This contains all of the logic
       for determining how and where to check for standard spelling errors, context spelling
       errors and getting recommendations. It is a little messy, but recognized as some
       technical debt that will be refactored in the next sprint as we move into the tool
       becoming a 'live' spell checker.
A new file was also created called testset_main.py. This is an executable python script
       that takes in arguments for the JSON DictDatabase, a test set (formatted like the
       ones provided by Dr. Scannell, and a language specifier. The logic within this script
       mirrors the logic in logicCntrl() but handles data in the form of a test set rather
       than a raw user input. I also want to both clean this file up going forward and add
       some metrics calculations to better asses improvements to the spell checker.
There is also an updated version of the English Dictionary in this commit. A small issue
       was found in the parser since the last database was built and so this updates corrects
       those errors.
The rest of the changes are small and should be relatively self explanatory.

### CODE REVIEW

**Comment by abeiler7** [comment]
The Tests are not running automatically on the PR for some reason, but here is the link to the passing tests from this branch in my fork of the repo.
https://github.com/abeiler7/SoftwareDevProject2021/actions/runs/1469988627

File: app/create_db.py [hunk 1]
```diff
@@ -1,5 +1,4 @@
-#!/usr/bin/env python
-
+import regex
```
Comment (GreihMurray) [inline app/create_db.py:1]: This takes us up to two imports of regex in the same file

File: app/spell_check.py [hunk 2]
```diff
@@ -40,7 +41,71 @@ def check_other_lang(input_list, word_list, dictionary):
     results = []
     for word_idx in word_list:
         if not regex.search(r'([^\p{L}\'\-])', input_list[word_idx]):
-            lower_word = str(unicodedata.normalize('NFC', input_list[word_idx])).lower()
+            lower_word = toLower(input_list[word_idx])
```
Comment (GreihMurray) [inline app/spell_check.py:44]: Is this working with Irish diacritics properly?

File: app/create_db.py (hunk 1 above)
Comment (abeiler7) [reply app/create_db.py:1]: Yeah that shouldn't be there! I may have messed that up while handling some merge conflicts as there were a lot in our import sections.

File: app/spell_check.py (hunk 2 above)
Comment (GreihMurray) [reply app/spell_check.py:44]: yep, I now realize that is not the built in lower function in python, sorry it was too early when I looked at this the first time apparently

**Review by GreihMurray — APPROVED** [review]
I'm going to go ahead and approve, just do me a favor and get that second import regex out prior to merging. It wont break anything but its also unnecessary

**Review by bmacgavin — APPROVED** [review]

**Pull request merged**